File: exercises/java/src/main/java/rename/Thingy.java
package rename;

import javax.xml.bind.DatatypeConverter;
import java.nio.charset.StandardCharsets;

class Thingy {

    private int foo;
    private int c1=0x1, d2=0b111;
    private int foobar = new int[] {0,0,0,0,0}.length;

    String doTheThing() {
        String s = "";
        for (; foo < Byte.MAX_VALUE - 27; foo++) s += b(foo) + " ";
        return s.substring(0,s.length()-1);
    }

    private String b(int foo) {
        c1++;
        d2--;
        foobar--;
        String s = c1 == d2||foobar==0?"":String.valueOf(foo + 1);
        if (c1 == d2) s += reset();
        if (foobar==0) s += f2();
        return s;
    }

    private String f2() {
        foobar = new int[] {0,0,0,0,0}.length;
        return new String(DatatypeConverter.parseHexBinary("42757a7a"), StandardCharsets.UTF_8);
    }

    private String reset() {
        d2 = 0x1B;
        c1 = 0x15;
        return new String(DatatypeConverter.parseHexBinary("46697a7a"), StandardCharsets.UTF_8);
    }
}

File: exercises/c-sharp/apprentice-bootcamp-fundamentals-2/apprentice-bootcamp-fundamentals-2/Thingy.cs
using System;

namespace apprentice_bootcamp_fundamentals_2
{
    public class Thingy
    {
        private int foo;
        private int bar;
        private int foobar = new int[] { 0, 0, 0, 0, 0 }.Length;

        public string DoTheThing()
        {
            string s = "";
            for (; foo < Byte.MaxValue - 27; foo++) s += b(foo) + " ";
            return s.Substring(0, s.Length - 1);
        }

        private string b(int foo)
        {
            bar++;
            foobar--;

            string s = bar == 0b11 || foobar == 0 ? "" : (foo + 1).ToString();
            if (bar == 0b11) s += BarFunction();
            if (foobar == 0) s += FooFunction();
            return s;
        }

        private string FooFunction()
        {
            foobar = new int[] { 0, 0, 0, 0, 0 }.Length;
            return DataTypeConverter.ParseHexBinary("42757a7a");
        }

        private string BarFunction()
        {
            bar = 0;
            return DataTypeConverter.ParseHexBinary("46697a7a");
        }
    }
}
